lnwallet/chanfunding/coin_select.go

```diff
@@ -143,16 +147,22 @@ func calculateFees(utxos []wallet.Coin, feeRate chainfee.SatPerKWeight,
 	return requiredFeeNoChange, requiredFeeWithChange, nil
 }
 
-// sanityCheckFee checks if the specified fee amounts to over 20% of the total
-// output amount and raises an error.
-func sanityCheckFee(totalOut, fee btcutil.Amount) error {
-	// Fail if more than 20% goes to fees.
-	// TODO(halseth): smarter fee limit. Make configurable or dynamic wrt
-	// total funding size?
-	if fee > totalOut/5 {
-		return fmt.Errorf("fee %v on total output value %v", fee,
-			totalOut)
+// sanityCheckFee checks if the specified fee amounts to what the provided ratio
+// allows.
+func sanityCheckFee(totalOut, fee btcutil.Amount, maxFeeRatio float64) error {
+	// Sanity check the maxFeeRatio itself.
+	if maxFeeRatio <= 0.00 || maxFeeRatio > 1.00 {
+		return fmt.Errorf("maxFeeRatio must be between 0.00 and 1.00 "+
```

Comment:
This change in error message requires a change in https://github.com/lightningnetwork/lnd/blob/e99e6662cfd53851e63b7bd53f4d211eac89ffd6/lnwallet/chanfunding/coin_select_test.go#L414 as well.